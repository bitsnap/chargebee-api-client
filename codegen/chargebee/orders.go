package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListOrders generates chargebee client code to fetch all orders
//
// API: https://{site}.chargebee.com/api/v2/orders
//
// Documentation: https://apidocs.chargebee.com/docs/api/orders?lang=curl#list_orders
func GenerateListOrders() string {
	return client.GenerateList("Orders", "Order", "chargebee.com/api/v2/orders")
}

// GenerateRetrieveOrder generates chargebee client code to retrieve specific order
//
// API: https://{site}.chargebee.com/api/v2/orders/{order-id}
//
// Documentation: https://apidocs.chargebee.com/docs/api/orders?lang=curl#retrieve_an_order
func GenerateRetrieveOrder() string {
	return client.GenerateRetrieve("Order", "chargebee.com/api/v2/orders")
}

// GenerateOrdersModel generates chargebee Order domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/orders?lang=curl#order_attributes
func GenerateOrdersModel() string {
	return client.GenerateModel("Order", "Order attributes", "https://apidocs.chargebee.com/docs/api/orders")
}
