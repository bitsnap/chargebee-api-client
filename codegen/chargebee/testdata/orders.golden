# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee orders code generation when generating Chargebee API client should be able to generate orders models" = '''
(string) (len=6401) "type Order struct {\n    Id string `json:\"id\" validate:\"required\"`\n    DocumentNumber string `json:\"document_number\"`\n    InvoiceId string `json:\"invoice_id\"`\n    SubscriptionId string `json:\"subscription_id\"`\n    CustomerId string `json:\"customer_id\"`\n    Status enums.StatusEnum `json:\"status\"`\n    CancellationReason enums.CancellationReasonEnum `json:\"cancellation_reason\"`\n    PaymentStatus enums.PaymentStatusEnum `json:\"payment_status\"`\n    OrderType enums.OrderTypeEnum `json:\"order_type\"`\n    PriceType enums.PriceTypeEnum `json:\"price_type\" validate:\"required\"`\n    ReferenceId string `json:\"reference_id\"`\n    FulfillmentStatus string `json:\"fulfillment_status\"`\n    OrderDate *time.Time `json:\"order_date\"`\n    ShippingDate *time.Time `json:\"shipping_date\"`\n    Note string `json:\"note\"`\n    TrackingId string `json:\"tracking_id\"`\n    TrackingUrl string `json:\"tracking_url\"`\n    BatchId string `json:\"batch_id\"`\n    CreatedBy string `json:\"created_by\"`\n    ShipmentCarrier string `json:\"shipment_carrier\"`\n    InvoiceRoundOffAmount uint64 `json:\"invoice_round_off_amount\"`\n    Tax uint64 `json:\"tax\"`\n    AmountPaid uint64 `json:\"amount_paid\"`\n    AmountAdjusted uint64 `json:\"amount_adjusted\"`\n    RefundableCreditsIssued uint64 `json:\"refundable_credits_issued\"`\n    RefundableCredits uint64 `json:\"refundable_credits\"`\n    RoundingAdjustement uint64 `json:\"rounding_adjustement\"`\n    PaidOn *time.Time `json:\"paid_on\"`\n    ShippingCutOffDate *time.Time `json:\"shipping_cut_off_date\"`\n    CreatedAt *time.Time `json:\"created_at\" validate:\"required\"`\n    StatusUpdateAt *time.Time `json:\"status_update_at\"`\n    DeliveredAt *time.Time `json:\"delivered_at\"`\n    ShippedAt *time.Time `json:\"shipped_at\"`\n    ResourceVersion int64 `json:\"resource_version\"`\n    UpdatedAt *time.Time `json:\"updated_at\"`\n    CancelledAt *time.Time `json:\"cancelled_at\"`\n    ResentStatus enums.ResentStatusEnum `json:\"resent_status\"`\n    IsResent bool `json:\"is_resent\" validate:\"required\"`\n    OriginalOrderId string `json:\"original_order_id\"`\n    Discount uint64 `json:\"discount\"`\n    SubTotal uint64 `json:\"sub_total\"`\n    Total uint64 `json:\"total\"`\n    Deleted bool `json:\"deleted\" validate:\"required\"`\n    CurrencyCode string `json:\"currency_code\"`\n    IsGifted bool `json:\"is_gifted\"`\n    GiftNote string `json:\"gift_note\"`\n    GiftId string `json:\"gift_id\"`\n    ResendReason string `json:\"resend_reason\"`\n    BusinessEntityId string `json:\"business_entity_id\"`\n    OrderLineItems []OrderLineItem `json:\"order_line_items\"`\n    Id string `json:\"id\" validate:\"required\"`\n    InvoiceId string `json:\"invoice_id\" validate:\"required\"`\n    InvoiceLineItemId string `json:\"invoice_line_item_id\" validate:\"required\"`\n    UnitPrice uint64 `json:\"unit_price\"`\n    Description string `json:\"description\"`\n    Amount uint64 `json:\"amount\"`\n    FulfillmentQuantity int `json:\"fulfillment_quantity\"`\n    FulfillmentAmount uint64 `json:\"fulfillment_amount\"`\n    TaxAmount uint64 `json:\"tax_amount\"`\n    AmountPaid uint64 `json:\"amount_paid\"`\n    AmountAdjusted uint64 `json:\"amount_adjusted\"`\n    RefundableCreditsIssued uint64 `json:\"refundable_credits_issued\"`\n    RefundableCredits uint64 `json:\"refundable_credits\"`\n    IsShippable bool `json:\"is_shippable\" validate:\"required\"`\n    Sku string `json:\"sku\"`\n    Status enums.StatusEnum `json:\"status\"`\n    EntityType enums.EntityTypeEnum `json:\"entity_type\" validate:\"required\"`\n    ItemLevelDiscountAmount uint64 `json:\"item_level_discount_amount\"`\n    DiscountAmount uint64 `json:\"discount_amount\"`\n    EntityId string `json:\"entity_id\"`\n    FirstName string `json:\"first_name\"`\n    LastName string `json:\"last_name\"`\n    Email string `json:\"email\"`\n    Company string `json:\"company\"`\n    Phone string `json:\"phone\"`\n    Line1 string `json:\"line1\"`\n    Line2 string `json:\"line2\"`\n    Line3 string `json:\"line3\"`\n    City string `json:\"city\"`\n    StateCode string `json:\"state_code\"`\n    State string `json:\"state\"`\n    Country string `json:\"country\"`\n    Zip string `json:\"zip\"`\n    ValidationStatus enums.ValidationStatusEnum `json:\"validation_status\"`\n    Index int `json:\"index\" validate:\"required\"`\n    FirstName string `json:\"first_name\"`\n    LastName string `json:\"last_name\"`\n    Email string `json:\"email\"`\n    Company string `json:\"company\"`\n    Phone string `json:\"phone\"`\n    Line1 string `json:\"line1\"`\n    Line2 string `json:\"line2\"`\n    Line3 string `json:\"line3\"`\n    City string `json:\"city\"`\n    StateCode string `json:\"state_code\"`\n    State string `json:\"state\"`\n    Country string `json:\"country\"`\n    Zip string `json:\"zip\"`\n    ValidationStatus enums.ValidationStatusEnum `json:\"validation_status\"`\n    LineItemTaxes []LineItemTax `json:\"line_item_taxes\"`\n    LineItemId string `json:\"line_item_id\"`\n    TaxName string `json:\"tax_name\" validate:\"required\"`\n    DateTo *time.Time `json:\"date_to\"`\n    DateFrom *time.Time `json:\"date_from\"`\n    ProratedTaxableAmount *big.Float `json:\"prorated_taxable_amount\"`\n    IsPartialTaxApplied bool `json:\"is_partial_tax_applied\"`\n    IsNonComplianceTax bool `json:\"is_non_compliance_tax\"`\n    TaxableAmount uint64 `json:\"taxable_amount\" validate:\"required\"`\n    TaxAmount uint64 `json:\"tax_amount\" validate:\"required\"`\n    TaxJurisType enums.TaxJurisTypeEnum `json:\"tax_juris_type\"`\n    TaxJurisName string `json:\"tax_juris_name\"`\n    TaxJurisCode string `json:\"tax_juris_code\"`\n    TaxAmountInLocalCurrency uint64 `json:\"tax_amount_in_local_currency\"`\n    LocalCurrencyCode string `json:\"local_currency_code\"`\n    LineItemDiscounts []LineItemDiscount `json:\"line_item_discounts\"`\n    LineItemId string `json:\"line_item_id\" validate:\"required\"`\n    DiscountType enums.DiscountTypeEnum `json:\"discount_type\" validate:\"required\"`\n    EntityId string `json:\"entity_id\"`\n    DiscountAmount uint64 `json:\"discount_amount\" validate:\"required\"`\n    LinkedCreditNotes []OrderLineItemLinkedCredit `json:\"linked_credit_notes\"`\n    Amount uint64 `json:\"amount\"`\n    Type enums.TypeEnum `json:\"type\" validate:\"required\"`\n    Id string `json:\"id\" validate:\"required\"`\n    Status enums.StatusEnum `json:\"status\" validate:\"required\"`\n    AmountAdjusted uint64 `json:\"amount_adjusted\"`\n    AmountRefunded uint64 `json:\"amount_refunded\"`\n    ResentOrders []ResentOrder `json:\"resent_orders\"`\n    OrderId string `json:\"order_id\" validate:\"required\"`\n    Reason string `json:\"reason\"`\n    Amount uint64 `json:\"amount\"`\n}"
'''
